TASK-4/EmployeeReportGenerator.cs

```diff
@@ -0,0 +1,32 @@
+namespace Persons;
+
+public class EmployeeReportGenerator : IReportGenerator
+{
+    public IEnumerable<Employee> ListOfEmployees;
+
+    public EmployeeReportGenerator(IEnumerable<Employee> listOfEmployees)
+    {
+        ListOfEmployees = listOfEmployees;
+    }
+
+    public void Report()
+    {
+        const string template = "{0,36} | {1,25} | {2, 40} | {3,8} | {4, -30}";
+        Console.WriteLine(template, "Id:", "Name Surname:", "Position:", "Salary:", "Company name:");
+
+        var orderedEmployees = from selectedEmployee in ListOfEmployees
+                               orderby selectedEmployee.CompanyName, selectedEmployee.Salary descending
+                               select selectedEmployee;
+        Employee employee;
```

Comment:
Remove this and better use var inside the loop

Reply:
Done.